web_company_color/models/res_company.py

```diff
@@ -0,0 +1,165 @@
+            }
+          }
+
+          > ul {
+            > li {
+              > a, > label {
+                color: %(color_navbar_text)s !important;
+
+                &:hover, &:focus, &:active, &:focus:active {
+                  background-color: %(color_navbar_bg_hover)s !important;
+                }
+              }
+            }
+          }
+        }
+    """
+
+    company_colors = fields.Serialized()
+    color_navbar_bg = fields.Char('Navbar Background Color',
+                                  sparse='company_colors')
+    color_navbar_bg_hover = fields.Char(
+        'Navbar Background Color Hover', sparse='company_colors')
+    color_navbar_text = fields.Char('Navbar Text Color',
+                                    sparse='company_colors')
+    scss_modif_hash = fields.Char('SCSS Modif. Hash')
+
+    @api.model
+    def create(self, values):
+        record = super(ResCompany, self).create(values)
+        record.create_or_update_scss_attachment()
+        return record
+
+    @api.multi
+    def unlink(self):
+        result = super(ResCompany, self).unlink()
+        IrAttachmentObj = self.env['ir.attachment']
+        for record in self:
+            IrAttachmentObj.sudo().search([
+                ('url', 'like', '%s%%' % record._get_url_scss_simplified()),
+            ]).sudo().unlink()
+        return result
+
+    @api.multi
+    def write(self, values):
+        if not self.env.context.get('ignore_company_color', False):
+            fields_to_check = ('color_navbar_bg',
+                               'color_navbar_bg_hover',
+                               'color_navbar_text')
+            if 'logo' in values:
+                if values['logo']:
+                    _r, _g, _b = image_to_rgb(convert_to_image(values['logo']))
+                    # Make color 10% darker
+                    _h, _l, _s = rgb_to_hls(_r, _g, _b)
+                    _l = max(0, _l - 0.1)
+                    _rd, _gd, _bd = hls_to_rgb(_h, _l, _s)
+                    # Calc. optimal text color (b/w)
+                    # Grayscale human vision perception (rec.709 values)
+                    _a = 1 - (0.2126 * _r + 0.7152 * _g + 0.0722 * _b)
+                    values.update({
+                        'color_navbar_bg': n_rgb_to_hex(_r, _g, _b),
+                        'color_navbar_bg_hover': n_rgb_to_hex(_rd, _gd, _bd),
+                        'color_navbar_text': '#000' if _a < 0.5 else '#fff',
+                    })
+                else:
+                    values.update(self.default_get(fields_to_check))
+
+            result = super(ResCompany, self).write(values)
+
+            if any([field in values for field in fields_to_check]):
+                self.create_or_update_scss_attachment()
+        else:
+            result = super(ResCompany, self).write(values)
+        return result
+
+    @api.multi
+    def _sanitize_scss_values(self, values):
+        self.ensure_one()
+        values = dict(self.company_colors)
```

Comment:
do you need to call `dict` here? I think you get an empty one anyway

Reply:
Fixed, thanks